### PR TITLE
[MRG+1] Updated tutorial

### PR DESCRIPTION
**Pull request by carlosp420**
The DMOZ pages have changed recently. Updated some text and code from
the tutorial so the example spider will work on the new layout.

### CODE REVIEW

File: docs/intro/tutorial.rst [hunk 1]
```diff
@@ -323,33 +323,35 @@ use Firefox Developer Tools or some Firefox extensions like Firebug. For more
 information see :ref:`topics-firebug` and :ref:`topics-firefox`.
 
 After inspecting the page source, you'll find that the web site's information
-is inside a ``<ul>`` element, in fact the *second* ``<ul>`` element.
+is inside ``<div>`` elements. In fact, the title and description of the site's list
```
Comment (redapple) [inline docs/intro/tutorial.rst]: I would go as far as showing a bit of the HTML:

```
                <div id="sites-section" class="section-wrapper">
                        <div id="site-list-content" class="results browse-content">
            
                            <div class="site-item ">
                                <div class="site-icon "><i class='siteIcon fa fa-file-text-o'></i></div>
                                <div class="site-flag">
                                  <a href="/public/flag?t=computers&cat=Computers%2FProgramming%2FLanguages%2FPython%2FBooks&url=http%3A%2F%2Fwww.pearsonhighered.com%2Feducator%2Facademic%2Fproduct%2F0%2C%2C0130260363%2C00%252Ben-USS_01DBC.html" 
                                     title="Report an issue with this listing"><i class='fa fa-flag'></i></a>
                                </div>

                                <div class="title-and-desc">
                                  <a target="_blank" href="http://www.pearsonhighered.com/educator/academic/product/0,,0130260363,00%2Ben-USS_01DBC.html">
                                    <div class="site-title">Core Python Programming </div>
                                  </a>
                                  <div class="site-descr ">
			
                                    By Wesley J. Chun; Prentice Hall PTR, 2001, ISBN 0130260363. For experienced developers to improve extant skills; professional level examples. Starts by introducing syntax, objects, error handling, functions, classes, built-ins. [Prentice Hall]
                                    <span class="content-type media-date"></span>
                                  </div>
                                </div>
                            </div>
```

File: docs/intro/tutorial.rst [hunk 2]
```diff
@@ -323,33 +323,35 @@ use Firefox Developer Tools or some Firefox extensions like Firebug. For more
...
+are included in ``<div>`` elements that contain the ``class`` attribute
+``"title-and-desc"``.
 
-So we can select each ``<li>`` element belonging to the site's list with this
+So we can select each ``<div>`` element belonging to the site's list with this
 code::
 
-    response.xpath('//ul/li')
+    response.xpath('//div[@class="title-and-desc"]')
```
Comment (redapple) [inline docs/intro/tutorial.rst]: I believe the equivalent to previous `//ul/li` is selecting the div with the list of site divs,
e.g. `response.xpath('//div[@id="site-list-content"]/div[@class="site-item "]')`

The CSS version of this is much simpler (as is worth a mention)
`response.css('#site-list-content > div.site-item')`

File: docs/intro/tutorial.rst [hunk 3]
```diff
@@ -323,33 +323,35 @@ use Firefox Developer Tools or some Firefox extensions like Firebug. For more
...
 
 And from them, the site's descriptions::
 
-    response.xpath('//ul/li/text()').extract()
+    response.xpath('//div[@class="title-and-desc"]/div/text()').extract()
```
Comment (redapple) [inline docs/intro/tutorial.rst]: I would prefer that the tutorial shows using specific elements classes, like `<div class="site-descr ">` here,
e.g. `response.xpath('//div[@id="site-list-content"]/div[@class="site-item "]/div[@class="title-and-desc"]/div[@class="site-descr "]')`
(although it becomes quite ugly -- a reason to show how to mix/chain XPath and CSS)

File: docs/intro/tutorial.rst [hunk 4]
```diff
@@ -323,33 +323,35 @@ use Firefox Developer Tools or some Firefox extensions like Firebug. For more
...
 
 The site's titles::
 
-    response.xpath('//ul/li/a/text()').extract()
+    response.xpath('//div[@class="title-and-desc"]/a/div/text()').extract()
```
Comment (redapple) [inline docs/intro/tutorial.rst]: titles also have their class now: `<div class="site-title">Core Python Programming </div>`

**Comment by kmike** [comment]
See also: https://github.com/scrapy/dirbot/pull/20. It'd be good to have similar code in both repos. 

`.xpath("//div[@class="title-and-desc"])` is longer and more brittle than `.css("div.title-and-desc")`: it won't work if there are extra classes; similar code also won't work if there are 2 classes which order can change randomly. I'm also not sure it works with extra spaces in attribute value (but haven't checked that).

**Comment by carlosp420** [comment]
hi @redapple I updated this PR.

File: docs/intro/tutorial.rst [hunk 5]
```diff
@@ -372,10 +407,10 @@ Let's add this code to our spider::
         ]
      
         def parse(self, response):
-            for sel in response.xpath('//ul/li'):
-                title = sel.xpath('a/text()').extract()
+            for sel in response.xpath('//div[@class="title-and-desc"]'):
```
Comment (redapple) [inline docs/intro/tutorial.rst]: The transition from `for sel in response.xpath('//div[@id="site-list-content"]/...` to `for sel in response.xpath('//div[@class="title-and-desc"]'):` is not explained.

File: docs/intro/tutorial.rst [hunk 6]
```diff
@@ -323,34 +323,69 @@ use Firefox Developer Tools or some Firefox extensions like Firebug. For more
...
+	response.xpath('//div[@id="site-list-content"]/div[@class="site-item "]')
+
+You can also use CSS selectors as the expression is much simpler::
+
+	response.css('#site-list-content > div.site-item')
 
 And from them, the site's descriptions::
 
-    response.xpath('//ul/li/text()').extract()
+	response.xpath('//div[@id="site-list-content"]/div[@class="site-item "]/div[@class="title-and-desc"]/div[@class="site-descr "]/text()').extract()
```
Comment (redapple) [inline docs/intro/tutorial.rst]: Here is a good case for introducing whitespaced XPaths, to bring a bit of air.
I tend to like triple-quoted strings, i.e:

```
response.xpath('''//div[@id="site-list-content"]
                        /div[@class="site-item "]
                            /div[@class="title-and-desc"]
                                /div[@class="site-descr "]
                                    /text()''').extract()
```

File: docs/intro/tutorial.rst [hunk 7]
```diff
@@ -323,34 +323,69 @@ use Firefox Developer Tools or some Firefox extensions like Firebug. For more
...
+
+Again, using CSS selectors is much simpler::
+
+	response.css('#site-list-content > div.site-item > div.title-and-desc > div.site-descr::text').extract()
```
Comment (redapple) [inline docs/intro/tutorial.rst]: Same as above, one can use:

```
response.css('''#site-list-content
                    > div.site-item
                        > div.title-and-desc
                            > div.site-descr::text''').extract()
```

**Comment by redapple** [comment]
hi @carlosp420 , sorry for the very late review. I left some comments in-line.

**Comment by redapple** [comment]
LGTM, thanks @carlosp420 !
what do you think @kmike , @eliasdorneles , @stummjr ?

**Comment by stummjr** [comment]
Hey folks!

I loved the introduction of CSS alternatives for XPath expressions in the tutorial.

I'm only concerned about the XPath expressions used to capture titles, descriptions and URLs. While they are quite precise, they might be kind of scary for newcomers trying out Scrapy for the first time. I mean, they are not complex, but big.

There's another PR to fix the DMOZ examples (https://github.com/scrapy/scrapy/pull/2180) that uses simpler XPath expressions. It has a [few errors](https://github.com/scrapy/scrapy/pull/2180/files#diff-e64b94efdcc21a322c82ccbfcd009a1cR350) though. 

I think we could have a mix of both PRs: including CSS selectors as this PR does and using the simpler XPath expressions from the other PR.

Thoughts?

**Comment by kmike** [comment]
Thanks @carlosp420! Your changes were good 👍, but we've rewritten the tutorial to use http://toscrape.com instead of dmoz (#2236), in order to prevent it from breaking again in future.

**Comment by carlosp420** [comment]
Doh!

**Pull request closed without merging**